### PR TITLE
ReturnLabel: module provides client to micro-service

### PR DESCRIPTION
**Pull request by DerCalli**
**Description**

module provides client to micro-service. Expects customer-id and  company-unit-address-uuid, returns base64 encoded pdf.

- zed layer
- unit-tests

### CODE REVIEW

File: bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Api/Adapter/ReturnLabelAdapter.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * @param \Spryker\Shared\Kernel\Transfer\AbstractTransfer $transfer
+     *
+     * @return array
+     */
+    protected function createOptions(AbstractTransfer $transfer): array
+    {
+        $options = [
+            RequestOptions::HEADERS => static::DEFAULT_HEADERS,
+            RequestOptions::TIMEOUT => 4,
```
Comment (julianzimmermann) [inline bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Api/Adapter/ReturnLabelAdapter.php:74]: Should be via config

File: bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Api/Adapter/ReturnLabelAdapter.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * @param array $options
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    protected function send(array $options = []): ResponseInterface
+    {
+        $uri = $this->getUri();
+
+        return $this->httpClient->request('POST', $uri, $options);
```
Comment (julianzimmermann) [inline bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Api/Adapter/ReturnLabelAdapter.php:98]: use const

File: bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Mapper/ReturnLabelCustomerMapper.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+     * @param \Generated\Shared\Transfer\ReturnLabelCustomerTransfer $returnLabelCustomerTransfer
+     *
+     * @return \Generated\Shared\Transfer\ReturnLabelCustomerTransfer
+     */
+    public function mapCompanyBusinessUnitToReturnLabelCustomer(
+        CompanyBusinessUnitTransfer $companyBusinessUnitTransfer,
+        ReturnLabelCustomerTransfer $returnLabelCustomerTransfer
+    ): ReturnLabelCustomerTransfer {
+        return $returnLabelCustomerTransfer
+            ->setReceiverId('deu')
```
Comment (julianzimmermann) [inline bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Mapper/ReturnLabelCustomerMapper.php:36]: const

File: bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Model/ReturnLabelGenerator.php [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+     * @param \FondOfOryx\Zed\ReturnLabel\ReturnLabelConfig $config
+     */
+    public function __construct(
+        CompanyUnitAddressReaderInterface $companyUnitAddressReader,
+        CompanyBusinessUnitReaderInterface $companyBusinessUnitReader,
+        ReturnLabelAdapterInterface $returnLabelAdapter,
+        ReturnLabelCustomerMapperInterface $returnLabelCustomerMapper,
+        ReturnLabelConfig $config
+    ) {
+        $this->companyUnitAddressReader = $companyUnitAddressReader;
```
Comment (julianzimmermann) [inline bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Model/ReturnLabelGenerator.php:56]: maybe reader collection or reader resource

File: bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/ReturnLabelBusinessFactory.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+ */
+class ReturnLabelBusinessFactory extends AbstractBusinessFactory
+{
+    /**
+     * @return \FondOfOryx\Zed\ReturnLabel\Business\Model\ReturnLabelGeneratorInterface
+     */
+    public function createReturnLabelGenerator(): ReturnLabelGeneratorInterface
+    {
+        return new ReturnLabelGenerator(
+            $this->createCompanyUnitAddressReader(),
```
Comment (julianzimmermann) [inline bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/ReturnLabelBusinessFactory.php:35]: as before

**Review by julianzimmermann — COMMENTED** [review]
see comments

File: bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Mapper/ReturnLabelCustomerMapper.php (hunk 3 above)
Comment (DerCalli) [reply bundles/return-label/src/FondOfOryx/Zed/ReturnLabel/Business/Mapper/ReturnLabelCustomerMapper.php:36]: better move to config :)))

**Pull request merged**